tests/core/utilities/test_lightning_cli.py

```diff
@@ -79,19 +79,19 @@ def test_add_argparse_args_redefined(cli_args):
         ("--auto_lr_find=True --auto_scale_batch_size=power", {"auto_lr_find": True, "auto_scale_batch_size": "power"}),
         (
             "--auto_lr_find any_string --auto_scale_batch_size ON",
-            {"auto_lr_find": "any_string", "auto_scale_batch_size": True},
```

Comment:
This was an unintended behavior in an old version of jsonargparse. For a union of types, in this case `Union[str, bool]`, the subtypes are validated from left to right. The first type that succeeds will be used. Since `str` is first, a string value will stay as a string. For strings to always be converted to booleans the type would need to be `Union[bool, str]`.

Reply:
:+1: